src/RadixDlt.NetworkGateway.PostgresIntegration/DbQueryExtensions.cs

```diff
@@ -70,7 +70,12 @@ namespace RadixDlt.NetworkGateway.PostgresIntegration;
 
 internal static class DbQueryExtensions
 {
-    public static IQueryable<LedgerTransaction> GetTopLedgerTransaction<TDbContext>(this TDbContext dbContext)
+    /// <summary>
+    /// The long/horrible/verbose name is on purpose, to draw your attention and prevent people foot-gunning themselves.
+    /// A LedgerTransaction row contains large blobs, so you must SELECT the fields you need after using this, and not pull down the whole
+    /// ledger transaction row, to avoid possible performance issues.
+    /// </summary>
+    public static IQueryable<LedgerTransaction> GetTopLedgerTransaction__EnsureYouSelectRequiredFieldsAsRowsCanBeVeryBig<TDbContext>(this TDbContext dbContext)
```

Comment:
maybe it'd be better to simply remove IQueryable extensions and create 2 speicfic methods, 1 would return state version and other summary?

Reply:
IMO this's not a good mechanism to prevent people from making that mistake as they can still run `dbContext.LedgerTransactions.Take(123).ToList()` and there's nothing to guarantee their safeness.
There's already this idea to move ledger TXs out of postgres db but that's a story for another day.

My point here is: this is effective one-time fix but it's neither going to scale nor prevent future issues.

EDIT: maybe a good intermediate solution would be to create a postgres view on LedgerTXs called `LightewigthLedgerTXs` where only few necessary columns would be available? This should work well with EF I think.